### PR TITLE
Support use with with-open. Add timeout for read-until.

### PR DESCRIPTION
**Pull request by cdzwm**

### CODE REVIEW

**Comment by komcrad** [comment]
I am aware this pull request has been open. I have not had time to review it yet. Hopefully I can get to it soon.

**Comment by cdzwm** [comment]
Ok




 
---Original---

File: src/clj_telnet/core.clj [hunk 1]
```diff
@@ -69,15 +74,11 @@
...
-      (.println s)
-      (.flush))))
-
-(defn with-telnet
-  [telnet f]
-  (try (f telnet)
-       (catch Exception e
-         (kill-telnet telnet)
-         (throw e))))
+  ([^TelnetClient telnet ^String s cr]
```
Comment (komcrad) [inline src/clj_telnet/core.clj]: This is completely changes the behavior of with-telnet.
I'm not a fan.

Comment (komcrad) [reply src/clj_telnet/core.clj]: It should probably be a macro that cleans up and throws the exception.

Comment (komcrad) [reply src/clj_telnet/core.clj]: https://github.com/komcrad/clj-telnet/pull/15/commits/26ddcae530c4ba1e894df6935ce2f8780ad2c393

This was a good commit. It makes with-telnet into what I would want.
I wanted something that would throw the exception. So adding with-open compatibility and re-writing with-telnet were reasonable changes.

**Pull request closed without merging**